core/contracts/financial-templates/implementation/FeePayer.sol

```diff
@@ -177,13 +177,26 @@ abstract contract FeePayer is Testable {
         return collateral.div(cumulativeFeeMultiplier);
     }
 
+    // Decrease rawCollateral by a fee-adjusted collateralToRemove amount. Fee adjustment scales up collateralToRemove
+    // by dividing it by cumulativeFeeMutliplier. There is potential for this quotient to be floored, therefore rawCollateral
+    // is decreased by less than expected. Because this method is usually called in conjunction with an actual removal of collateral
+    // from this contract, return the fee-adjusted amount that the rawCollateral is decreased by so that the caller can
+    // minimize error between collateral removed and rawCollateral debited.
     function _removeCollateral(FixedPoint.Unsigned storage rawCollateral, FixedPoint.Unsigned memory collateralToRemove)
         internal
+        returns (FixedPoint.Unsigned memory removedCollateral)
     {
+        FixedPoint.Unsigned memory initialBalance = _getCollateral(rawCollateral);
         FixedPoint.Unsigned memory adjustedCollateral = _convertCollateral(collateralToRemove);
         rawCollateral.rawValue = rawCollateral.sub(adjustedCollateral).rawValue;
+        removedCollateral = initialBalance.sub(_getCollateral(rawCollateral));
     }
 
+    // Increase rawCollateral by a fee-adjusted collateralToRemove amount. Fee adjustment scales up collateralToRemove
+    // by dividing it by cumulativeFeeMutliplier. There is potential for this quotient to be floored, therefore rawCollateral
+    // is increased by less than expected. Because this method is usually called in conjunction with an actual addition of collateral
+    // into this contract, there is no need to return the fee-adjusted amount that the rawCollateral is increased by because the contract
+    // will have surplus collateral in case an error is introduced.
     function _addCollateral(FixedPoint.Unsigned storage rawCollateral, FixedPoint.Unsigned memory collateralToAdd)
```

Comment:
Drive-by-comment: why not make this symmetric and do the same thing when we take collateral *in* to reduce rounding there too?

Reply:
So I definitely debated this but the tradeoff for me was practicality versus symmetry.

Symmetrically it makes sense to do the same thing when taking collateral in (`safeTransferFrom`) and calling `_addCollateral`.

Practically, we are OK if there is more collateral transferred in then added to `totalPositionCollateral`.

What do you think?

Reply:
While I totally agree that practically, it doesn't help much, I think the consistency and symmetry makes the code easier to read and understand. Obviously, if this adds a ton more complexity, then maybe it's not worth it, but IMO, it would be better just to do it in both directions so it's clearer that these operations are exactly opposite of one another.

Reply:
I've added the return value to `_addCollateral` but I'm still undecided about whether we should actually use that return value as input in `safeTransferFrom` for deposits/creates.

The problem with this is that sometimes the `totalPositionCollateral` is increased by 0 due to precision loss. Normally we'd still transfer 1 collateral from the user, but if we make a change here then its possible for deposits to transfer 0. However, if the method does not revert, then the `rawTotalPositionCollateral` will increase, but it will not reflect an increase in the `totalPositionCollateral`. It's a really odd case, but maybe not a bug.